### PR TITLE
Configuration support

### PR DESCRIPTION
**Pull request by wolf31o2**

### CODE REVIEW

File: attributes/config.rb [hunk 1]
```diff
@@ -27,3 +27,21 @@
...
+default['impala']['config']['IMPALA_STATE_STORE_ARGS'] = ' -log_dir=${IMPALA_LOG_DIR} -state_store_port=${IMPALA_STATE_STORE_PORT}'
+default['impala']['config']['IMPALA_SERVER_ARGS'] = ' \
+    -log_dir=${IMPALA_LOG_DIR} \
+    -catalog_service_host=${IMPALA_CATALOG_SERVICE_HOST} \
+    -state_store_port=${IMPALA_STATE_STORE_PORT} \
+    -use_statestore \
+    -state_store_host=${IMPALA_STATE_STORE_HOST} \
+    -be_port=${IMPALA_BACKEND_PORT}'
+default['impala']['config']['ENABLE_CORE_DUMPS'] = 'false'
+
```
Comment (dereklwood) [inline attributes/config.rb]: blank line: rubocop is angry

File: attributes/config.rb [hunk 2]
```diff
@@ -27,3 +27,21 @@
...
 override['hadoop']['distribution'] = 'cdh'
 override['hadoop']['distribution_version'] = 5
+
+# Impala /etc/default/impala startup configurations
+default['impala']['config']['IMPALA_CATALOG_SERVICE_HOST'] = '127.0.0.1'
+default['impala']['config']['IMPALA_STATE_STORE_HOST'] = '127.0.0.1'
+default['impala']['config']['IMPALA_STATE_STORE_PORT'] = '24000'
+default['impala']['config']['IMPALA_BACKEND_PORT'] = '22000'
+default['impala']['config']['IMPALA_LOG_DIR'] = '/var/log/impala'
+default['impala']['config']['IMPALA_CATALOG_ARGS'] = ' -log_dir=${IMPALA_LOG_DIR} '
```
Comment (dereklwood) [inline attributes/config.rb]: extra trailing space intended?

File: recipes/config.rb [hunk 3]
```diff
@@ -27,6 +27,56 @@
...
+  action :create
+  recursive true
+end
+
+# Setup /etc/default/impala
+if node['impala'].key?('config')
+  my_vars = { :options => node['impala']['config'] }
+
+  impala_log_dir =
+    if node['impala']['config'].key?('impala_log_dir')
```
Comment (dereklwood) [inline recipes/config.rb]: checking for lower case here.  default attributes are upper case (attributes are case-sensitive i thought?).  template does upcase anyway, so should probably just be consistent.

**Comment by dereklwood** [comment]
couple minor things otherwise :+1:

File: recipes/config.rb (hunk 3 above)
Comment (wolf31o2) [inline recipes/config.rb]: These should be lowercase, to match other cookbooks.

File: attributes/config.rb (hunk 2 above)
Comment (duddun) [inline attributes/config.rb]: Which trailing space?  I am not seeing a trailing space after any "\".
To which space are you referring?

File: recipes/config.rb (hunk 3 above)
Comment (duddun) [inline recipes/config.rb]: So in the attribute file, they should look like:
default['impala']['config']['impala_catalog_service_host'] = '127.0.0.1'
and this: templates/default/generic-env.sh.erb:  export <%= key.upcase %>="<%= value %>"
converts them to upper case in the resulting chef run, right?

Comment (dereklwood) [inline recipes/config.rb]: correct

**Pull request merged**